components/experimental/src/duration/duration.rs

```diff
@@ -0,0 +1,31 @@
+#[derive(Debug, Default, Clone, PartialEq, Eq, PartialOrd, Ord)]
+/// Represents a duration of time.
+pub struct Duration {
```

Comment:
As you can see in temporal duration: https://github.com/tc39/proposal-temporal/blob/main/docs/duration.md, they are supporting negative durations.

this means, we need to add a `sign` field that is `positive` by default.